ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/CanHardFork.hs

```diff
@@ -687,7 +687,7 @@ instance CardanoHardForkConstraints c => LedgerTablesCanHardFork (CardanoEras c)
       byron = InjectLedgerTables $ \NoByronLedgerTables -> polyEmptyLedgerTables
 
       shelley ::
-           forall era proto. SL.Crypto era ~ c
+           forall era proto. (SL.Crypto era ~ c, Eq (Core.TxOut era))
```

Comment:
Why do we need this constraint now? We should check with the ledger that this is not too strong.

Reply:
I think it can be removed because we don't perform the anti-diff sanity checks anymore

Reply:
Again, I stand corrected. See https://github.com/input-output-hk/ouroboros-network/pull/4036#discussion_r991936963

Reply:
> Why do we need this constraint now? We should check with the ledger that this is not too strong.

We need this constraint now because of `mapValuesAppliedMK`, which needs `Eq` specifically in the case that it is mapping over a `SeqDiffMK`

Reply:
@jorisdral I do not think that having the constraint will cause us any problems. Ledger has the same constraint in several places, such as the `EraTxOut` class.